### ISSUE TITLE
TechDraw: Area in mm² is not convenient (Add the ability to change the units of area measurement)

### ISSUE THREAD

**Issue by Pendelf** [FreeCAD/FreeCAD#18529]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

When measuring area, the global unit of measure mm2 is used, and as a rule a value in m2 is required. It may be worth adding the ability to select the area units in the measurement properties and/or TechDraw settings

![Снимок экрана 2024-12-15 225401](https://github.com/user-attachments/assets/14988119-888c-49b1-a56c-ab573a5b3102)

### Full version info

```shell
OS: Windows 11 build 26100
Architecture: x86_64
Version: 1.1.0dev.39646 (Git) Conda
Build type: Release
Branch: main
Hash: b25b83c1a58bf0ae95cbd502d0ccae27e8127cad
Python 3.11.11, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, OCC 7.8.1
Locale: Russian/Russia (ru_RU)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/Fusion
Installed mods: 
  * Curves 0.6.52
  * fasteners 0.5.33
  * PieMenu 1.10.2
  * sheetmetal 0.6.2
```

### Subproject(s) affected?

TechDraw

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by kaiwas** [comment]
I would like to add that the ability to change units of measurement in the property would be useful not only in Tech Draw, but also in other places.
Global settings of units of measurement are good, but very often you have to combine different units of measurement, but now there is no such option. We are limited to the global choice.

**Comment by pitch-circle** [comment]
Every dimension object in TechDraw has a getRawValue() method (send the dimension to the Python console and do obj.getRawValue() ).

Unfortunately, the result of this is not available in the Data panel.

If it were, it could be used in an Expression for the Format Spec of the displayed dimension.

That way, you could have an area in square millimetres and divide it by 1e^6 in the Format to see square metres.

It would allow far greater user control over combining units - among other things, it would allow very easy dual imperial/metric dimensioning.

**Comment by WandererFan** [comment]
This needs a read-only Property added to Dimension (called RawValue?) and mods to DrawViewDimensionPy.xml and DrawViewDimensionPyImp.cpp to add the attribute and a getter that calls getRawValue.

Then the raw value would be accessible to expressions.  Still have to handle the conversion from internal mm to the current units schema.

**Comment by grinapo** [comment]
Please help me understand what this issue is about! 
My problem is that when unit system is set to __Building Euro, cm, m², m³__ then linear measurements are correctly in cm, but area is in cm² instead of the set m². Does this issue cover that bug?

(Do I guess right that this should not stayed unfixed for v1.0.0?)

```
OS: Debian GNU/Linux forky/sid (XFCE/xfce/xcb)
Architecture: x86_64
Version: 1.0.0.
Build type: Release
Python 3.13.5, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, OCC 7.8.1
Locale: C/Default (C)
Stylesheet/Theme/QtStyle: FreeCAD Light.qss/FreeCAD Light/Fusion
```

**Comment by luzpaz** [comment]
> (Do I guess right that this should not stayed unfixed for v1.0.0?)

Yes, this most definitely will not make it into 1.0.x or the upcoming 1.1 (since we already are in a feature freeze). 

@Pendelf would you be so kind to update the ticket name to something more descriptive ?

**Comment by grinapo** [comment]
(I don't think it's about _convenience_: the actual unit system claims that areas **are in m²**. Also I believe there is three separate things were mentioned here: 
- when the unit system claims areas are in a specific unit then they should be (this really should be fixed),
- would be nice to be able to override the units separately (I guess that's what #22883 is about, i'ts been told that it's not that simple)
- would be nice to be able to override dimensions separately (I am not sure how feasible this is at all).

[Though I am, as a new user, _extremely_ confused by the unit system, since, for example, font sizes seem to be calculated by the units but limited to one decimals, so for a floor plan they cannot be set {in meters!} without switching the unit system to and from.])

**Comment by luzpaz** [comment]
@grinapo units system saw more improvements in 1.1.0dev  
Can you give it a test run ?

**Comment by WandererFan** [comment]
This one is fixed?  Don't know the PR that did it.

<img width="1611" height="757" alt="Image" src="https://github.com/user-attachments/assets/d4573758-d5a9-4e94-a88a-9f8a27603ee7" />

**Comment by grinapo** [comment]
> [@grinapo](https://github.com/grinapo) units system saw more improvements in 1.1.0dev Can you give it a test run ?

Is there anything in particular I shall look out for? It seems to work the same way.

**Comment by WandererFan** [comment]
Did we ever check the document level unit schema vs the system level schema here?

**Comment by grinapo** [comment]
@luzpaz If you meant setting the font sizes for example then yes, now it seems to be possible to enter arbitrarily precise numbers (as well as setting manually the units in mm, as now I have realised), but it still displayed as the set unit, with one decimal precision, so, for example, I get them in **cm**, and I can set it to **0.1 mm** but it gets displayed as **0.0 cm** (while it seems to be handled properly as 0.1 mm) and that's also the value (0.0 cm) offered for editing it. 
But this is possibly offtopic in this issue, so I apologise.

**Comment by luzpaz** [comment]
> Did we ever check the document level unit schema vs the system level schema here?
